src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -122,14 +124,46 @@ class BookingService(
   @Value("\${url-templates.frontend.application}") private val applicationUrlTemplate: String,
   @Value("\${url-templates.frontend.booking}") private val bookingUrlTemplate: String,
   @Value("\${arrived-departed-domain-events-disabled}") private val arrivedAndDepartedDomainEventsDisabled: Boolean,
+  private val userService: UserService,
+  private val userAccessService: UserAccessService,
+  private val bookingTransformer: BookingTransformer,
 ) {
   val approvedPremisesBookingAppealedCancellationReasonId: UUID =
     UUID.fromString("acba3547-ab22-442d-acec-2652e49895f2")
 
   private val log = LoggerFactory.getLogger(this::class.java)
 
   fun updateBooking(bookingEntity: BookingEntity): BookingEntity = bookingRepository.save(bookingEntity)
-  fun getBooking(id: UUID) = bookingRepository.findByIdOrNull(id)
+
+  fun getBooking(id: UUID): AuthorisableActionResult<Booking> {
+    val booking = bookingRepository.findByIdOrNull(id)
+      ?: return AuthorisableActionResult.NotFound("Booking", id.toString())
+
+    val user = userService.getUserForRequest()
+
+    if (!userAccessService.userCanManagePremisesBookings(user, booking.premises)) {
+      return AuthorisableActionResult.Unauthorised()
+    }
+
+    val personInfo = offenderService.getInfoForPerson(booking.crn, user.deliusUsername, user.hasQualification(UserQualification.LAO))
+
+    val staffMember = booking.keyWorkerStaffCode?.let { keyWorkerStaffCode ->
+      val premises = booking.premises
+
+      // Bookings will need to be specialised in a similar way to Premises so that TA Bookings do not have a keyWorkerStaffCode field
+      check (premises is ApprovedPremisesEntity) { "Booking has a Key Worker specified but Premises is not an ApprovedPremises" }
+
+      val staffMemberResult = staffMemberService.getStaffMemberByCode(keyWorkerStaffCode, premises.qCode)
+
+      if (staffMemberResult !is AuthorisableActionResult.Success) {
+        throw InternalServerErrorProblem("Unable to get Key Worker via Staff Code: $keyWorkerStaffCode / Q Code: ${premises.qCode}")
+      }
+
+      staffMemberResult.entity
+    }
+
+    return AuthorisableActionResult.Success(bookingTransformer.transformJpaToApi(booking, personInfo, staffMember))
```

Comment:
I believe this transformation is to build the API response, so am not sure this should happen in the service. Service should return agnostic response and controller should transform based on the respective API structure.

Reply:
Yes, really good point. I did quickly check this because i couldn't recall where transformers sit and I got unlucky in that i seemingly found the only service (ApplicationService) that uses transformers ... I'll make a note to move those too in the future! I've updated the code to move the transformer into the controller.